File: tests/integration/features/file_traceability/file_roles/01_relations_with_role/file.py
class Foo:
    """@relation(REQ-001, scope=class, role=Implementation)"""

    pass


def foo():
    """@relation(REQ-001, scope=function, role=Implementation)"""

    pass


# @relation(REQ-001, scope=range_start, role=Implementation)
def bar1():
    pass


def bar2():
    pass
# @relation(REQ-001, scope=range_end)


if __name__ == "__main__":
    pass  # @relation(REQ-001, scope=line, role=Implementation)

File: tests/integration/features/source_code_traceability/_file_roles/01_relations_with_role/file.c
// @relation(REQ-001, scope=function, role=Implementation)
extern int extfoo();

int extfoo() {
    return 0;
}

static void foo();

int main() {
    foo();  // @relation(REQ-001, scope=line, role=Implementation)
    return 0;
}

// @relation(REQ-001, scope=function, role=Implementation)
static void foo() {}

// @relation(REQ-001, scope=range_start, role=Implementation)
static void bar1() {}
static void bar2() {}
// @relation(REQ-001, scope=range_end)
